### DISCUSSION TITLE
How to build APIs

### DISCUSSION THREAD

**Discussion by justinmoon** [fedimint/fedimint#1190] [General]
TODO ... let's discuss

**Comment by dpc** [comment]
Seems like we want:

* push notification (I'm not sure if we really do, but it was mentioned)
* efficient encoding (in particular of larger binary data)
* ergonomic
* interoperability with other languages nice to have

**Reply by elsirion** [reply]
> efficient encoding

Not only efficient CPU/memory-wise, but also with minimal boilerplate. Having a struct

```rust
struct ProtobufFoo
    pub_key: [u8; 32]
    …
}
```

and then needing to manually and fallibly convert to

```rust
struct Foo {
    pub_key: PublicKey
    …
}
```

would be rather inconvenient.
